src/documents/tutorial-usingStringTemplates/UsingStringTemplates.md

```diff
@@ -0,0 +1,296 @@
+---
+title: Using String Templates
+layout: default
+---
+
+# Using String Templates #
+
+In the Infusion [Preference Framework](../PreferencesFramework.md), a "string template" is a mechanism for resolving variables within a string.
+
+This tutorial covers why you might use string templates, and gives a basic example of how to use string templates to:
+
+1.  Define "templates" (strings containing variables)
+2.  Define "terms", variables that will be replaced in templates.
+3.  Override "terms" and "templates" from a child component.
+
+This tutorial assumes that you are already familiar with [Infusion](../tutorial-gettingStartedWithInfusion/GettingStartedWithInfusion.md), and in particular with:
+1. [Defining components](../tutorial-gettingStartedWithInfusion/BasicComponentCreation-LittleComponents.md)
+2. [Defining invokers](../Invokers.md)
+3. [Defining and using expanders](../ExpansionOfComponentOptions.md)
+
+## Why do we need "string templates"?
+
+Let's talk about a common use case, in which we want to describe the location of files (configuration files, etc.) within a component's options.  If we have a single file location, a full path is relatively easy to update.
+
+If we have a long list of paths, people who wish to extend or even configure our component must laboriously copy the full list of file paths and customize each one.  This is likely to lead to "copy and paste" errors and unexpected behavior.  It would be better if users could specify their own home directory and have all of our file paths work relative to that new location.
+
+String templates give us the ability to describe things like file paths using values like `%myPath/myFilename`.  The percent values are placeholders for variable content that is replaced using a call to `fluid.stringTemplate` with the right options.
+
+## Writing a component that uses `fluid.stringTemplate`.
+
+Here is a simple component that has a `templates` block (containing strings with variables) and a `terms` block (containing a map of variable names and replacement values).  These names are arbitrary, you can name each of these whatever is most meaningful for you component.
```

Comment:
- although we recommend that you stick with these conventions unless you have a good reason to diverge.

typo you component -> your